storage/src/main/java/tech/pegasys/teku/storage/api/Eth1DepositChannel.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.api;
+
+import com.google.common.primitives.UnsignedLong;
+import tech.pegasys.teku.datastructures.blocks.Eth1BlockData;
+import tech.pegasys.teku.datastructures.operations.DepositWithIndex;
+
+public interface Eth1DepositChannel {
+  /**
+   * Called each time an eth1 deposit is added
+   *
+   * @param depositWithIndex the deposit to add to cache
+   */
+  void addEth1Deposit(final DepositWithIndex depositWithIndex);
+
+  void addEth1BlockData(final UnsignedLong timestamp, final Eth1BlockData eth1BlockData);
```

Comment:
Is there a reason these are two separate events?  We need to ensure that we store all deposits from a block as a single transaction or it will get very confusing knowing where the PowService should start from when reloading.

We could just have:
```
addEth1Deposits(final DepositsFromBlockEvent deposits)
```
That event actually contains everything we'd need to store and we could then batch the disk writes and store them all in one transaction.  We might want to use a different event but with similar data rather than passing the `pow.event.Deposit` instances through and making the storage system convert to `DepositWithIndex`